libra_toolbox/neutronics/neutron_source.py

```diff
@@ -0,0 +1,78 @@
+# building the MIT-VaultLab neutron generator
+# angular and energy distribution
+
+from pathlib import Path
+from collections.abc import Iterable
+import pandas as pd
+import numpy as np
+import h5py
+import openmc
+
+
+def A325_generator_diamond(center=(0, 0, 0), reference_uvw=(0, 0, 1)) -> Iterable[openmc.Source]:
+    """
+    Builds the MIT-VaultLab A-325 neutron generator in OpenMC
+    with data tabulated from John Ball and Shon Mackie characterization
+    via diamond detectors
+
+    Parameters
+    ----------
+    center : tuple, optional
+        coordinate position of the source (it is a point source),
+        by default (0, 0, 0)
+    reference_uvw : tuple, optional
+        direction for the polar angle (tuple or list of versors)
+    it is the same for the openmc.PolarAzimuthal class
+    more specifically, polar angle = 0 is the direction of the D accelerator
+    towards the Zr-T target, by default (0, 0, 1)
+
+    Returns
+    -------
+    Iterable[openmc.Source]
+        list of openmc neutron sources with angular and energy distribution 
+        and total strenght of 1
+    """
+
+    filename = "A325_generator_diamond.h5"
+    filename = str(Path(__file__).parent) / Path(filename)
+
+    with h5py.File(filename, "r") as source:
+        df = pd.DataFrame(source["values/table"][()]).drop(columns='index')
+        # energy values
+        energies = np.array(df["Energy (MeV)"]) * 1e6
+        # angle column names
+        angles = df.columns[1:]
+        # angular bins in [0, pi)
+        pbins = np.cos([np.deg2rad(float(a)) for a in angles] + [np.pi])
+        spectra = [np.array(source[col]) for col in angles]
```

Comment:
I believe this should be:
```suggestion
        spectra = [np.array(df[col]) for col in angles]
```